src/tools/miri/src/borrow_tracker/tree_borrows/exhaustive.rs

```diff
@@ -14,6 +14,7 @@ macro_rules! precondition {
         }
     };
 }
+#[cfg_attr(not(bootstrap), allow(unclear_local_imports))]
 pub(crate) use precondition;
```

Comment:
It seems like these re-exports of local macros cannot be disambiguated with either `self` nor `crate`, so the only way to fix the warning is to `allow` it...

Reply:
could you instead disable the lint for macro definitions?

Reply:
Yeah that's probably the best option. Though I don't think I quite understand where these macros even live then if they are neither in `self` nor in `crate` nor in `::`... Cc @petrochenkov

Reply:
Ideally this lint would work in `rustc_resolve` in roughly the same way as the redundant import lint (https://github.com/rust-lang/rust/pull/123813).
Then we'd check that `self::path` resolves to the same thing as `path` in all namespaces and it would work correctly in all cases.

Reply:
> Ideally this lint would work in rustc_resolve in roughly the same way as the redundant import lint (https://github.com/rust-lang/rust/pull/123813).

Hm... I am entirely unfamiliar with that part of the codebase. I'm happy for someone else to take over this PR, but it may be a while until I have the time to dig into this.

Reply:
I now made the lint just skip over macro imports.